src/runtime_src/xrt/device/hal2.cpp

```diff
@@ -484,13 +525,21 @@ read(const BufferObjectHandle& boh, void* dst, size_t sz, size_t offset, bool as
 }
 
 event
-device::sync(const BufferObjectHandle& boh, size_t sz, size_t offset, direction dir1, bool async)
+device::
+sync(const BufferObjectHandle& boh, size_t sz, size_t offset, direction dir1, bool async)
 {
   xclBOSyncDirection dir = XCL_BO_SYNC_BO_TO_DEVICE;
   if(dir1 == direction::DEVICE2HOST)
     dir = XCL_BO_SYNC_BO_FROM_DEVICE;
 
   BufferObject* bo = getBufferObject(boh);
+  if (bo->nodma) {
+    // sync is M2M copy between host and device handle
+    if (dir == XCL_BO_SYNC_BO_TO_DEVICE)
+      return event(typed_event<int>(m_ops->mCopyBO(m_handle, bo->handle, bo->nodma_host_handle, sz, offset, offset)));
```

Comment:
Perhaps this works:
```suggestion
      return { typed_event<int>(m_ops->mCopyBO(m_handle, bo->handle, bo->nodma_host_handle, sz, offset, offset)) };
```

Reply:
Yeah, sure.  This file is old and will be rewritten to use native APIs, so  will not bother with any style changes.